src/components/MapView/index.js

```diff
@@ -14,8 +14,33 @@ const MapView = ({ locations }) => {
   });
 
   const unknown = [];
+  const markers = [];
+
+  // maps coordinates to the locations at that coordinate
+  const coordsToLocationMap = new Map();
   locations.forEach((location) => {
-    if (!location.coords) unknown.push(location);
+    if (!location.coords) {
+      unknown.push(location);
+    } else if (coordsToLocationMap.has(location.coords)) {
+      coordsToLocationMap.get(location.coords).push(location);
+    } else {
+      coordsToLocationMap.set(location.coords, [location]);
+    }
+  });
+
+  coordsToLocationMap.forEach((coordLocations, coords) => {
```

Comment:
[nit] It would be more natural to use map here; i.e:

```js
const markers = coordsToLocationMap.entries().map(([coordLocations, cords]) =>
  // ...
);
```